### PR TITLE
cherry-pick RemoteExec transformer hotfix commits

### PR DESCRIPTION
**Pull request by alextheimer**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

### CODE REVIEW

File: query/src/main/scala/filodb/query/exec/MetadataRemoteExec.scala [hunk 1]
```diff
@@ -35,11 +34,11 @@ case class MetadataRemoteExec(queryEndpoint: String,
...
 
   private val dummyQueryStats = QueryStats()
-  override def sendHttpRequest(execPlan2Span: Span, httpTimeoutMs: Long)
-                              (implicit sched: Scheduler): Future[QueryResponse] = {
+  override def sendRequest(execPlan2Span: Span, httpTimeoutMs: Long)
+                          (implicit sched: Scheduler): Task[QueryResponse] = {
     import PromCirceSupport._
     import io.circe.parser
-    remoteExecHttpClient.httpMetadataPost(queryEndpoint, httpTimeoutMs,
+    val fut = remoteExecHttpClient.httpMetadataPost(queryEndpoint, httpTimeoutMs,
```
Comment (srivik) [inline query/src/main/scala/filodb/query/exec/MetadataRemoteExec.scala:41]: nit: more meaningful var name, "future" maybe?

File: query/src/main/scala/filodb/query/exec/PromQlRemoteExec.scala [hunk 2]
```diff
@@ -44,12 +43,12 @@ case class PromQlRemoteExec(queryEndpoint: String,
...
 
-  override def sendHttpRequest(execPlan2Span: Span, httpTimeoutMs: Long)
-                              (implicit sched: Scheduler): Future[QueryResponse] = {
+  override def sendRequest(execPlan2Span: Span, httpTimeoutMs: Long)
+                          (implicit sched: Scheduler): Task[QueryResponse] = {
 
     import PromCirceSupport._
     import io.circe.parser
-    remoteExecHttpClient.httpPost(queryEndpoint, requestTimeoutMs,
+    val fut = remoteExecHttpClient.httpPost(queryEndpoint, requestTimeoutMs,
```
Comment (srivik) [inline query/src/main/scala/filodb/query/exec/PromQlRemoteExec.scala:51]: nit: more meaningful var name, "future" maybe?

**Review by srivik — APPROVED** [review]
lgtm

File: query/src/main/scala/filodb/query/exec/PromQlRemoteExec.scala (hunk 2 above)
Comment (alextheimer) [reply query/src/main/scala/filodb/query/exec/PromQlRemoteExec.scala:51]: These are cherry-picked commits -- I'll leave these as-is

**Pull request merged**